cli/azd/cmd/util.go

```diff
@@ -331,3 +333,19 @@ func ensureProject(path string) error {
 
 	return nil
 }
+
+type envFlag struct {
+	environmentName string
+}
+
+func (e *envFlag) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	// Set the default value to AZURE_ENV_NAME value if the flag wasn't explicitly passed
+	e.environmentName = os.Getenv(environment.EnvNameEnvVarName)
+
+	local.StringVarP(
```

Comment:
I think this call has the side effect of writing the default value into the target, so it will stomp over the call to `os.Getenv`.

StringVarP calls `newStringValue`

https://github.com/spf13/pflag/blob/2e9d26c8c37aae03e3f9d4e90b7116f5accb7cab/string.go#L51-L54

It looks like `newStringValue` does the write here:

https://github.com/spf13/pflag/blob/2e9d26c8c37aae03e3f9d4e90b7116f5accb7cab/string.go#L51-L54

Reply:
We could just move the call to `os.Getenv` to L349, I guess?

Reply:
That's a good catch, and a bad miss by me. 

We could either have it inside `StringVarP` as a default value, which will materialize in the help text if the env variable is defined, or move it post L349 as you suggested, which will avoid the help text from showing up.

Personally, I think it's fine for the help text to show up as `(default: <value of AZURE_ENV_NAME>)` if you did define the env variable, so I went with that approach. Let me know if you think otherwise.

Reply:
> Personally, I think it's fine for the help text to show up as (default: <value of AZURE_ENV_NAME>) if you did define the env variable, so I went with that approach. Let me know if you think otherwise.

I think this is the right call, this is what my gut thought as well.